### ISSUE TITLE
handling of inplace operations

### ISSUE THREAD

**Issue by jnorwood** [KhronosGroup/NNEF-Tools#37]
Is there some convention for specifying in-place operations in the NNEF network definitions?  For example, I see a suffix of _tmp appended to the relu operation result variables in the Squeezenet conversion, and in  Squeezenet implementations I normally see relu implemented as in-place following the convolution, as in the example below, rather than allocating new output tensors.   I don't see any mention of _tmp suffix in the spec.

```
fire2_squeeze1x1 = conv(...)
fire2_squeeze1x1_tmp = relu(fire2_squeeze1x1)
```

**Comment by gyenesvi** [comment]
No, you cannot specify in-placing, as it is the private optimization matter of the inference engine, and it should not be controlled by the designer of the network.

The _tmp suffix is just the result of some graph transformation that was performed by the exporter.

**Comment by jnorwood** [comment]
I noticed that pytorch  adopted a convention to specify in-place operations.
https://discuss.pytorch.org/t/what-is-in-place-operation/16244

**Comment by gyenesvi** [comment]
It makes sense for PyTorch, since when you are writing a script, you are programming a runtime, that executes your operations, and an in-place operation makes a difference. However, NNEF is a file format, not a run-time, and when you just describe a network of computations, in-placing is not really useful, as it does not change the semantics of the computation. Furthermore, NNEF is like a functional language, where identifiers are only assigned once and are not mutable, which would be violated by in-place operations.
